src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -46,11 +47,12 @@ import Logger from 'src/utils/Logger'
 import { walletAddressSelector } from 'src/web3/selectors'
 
 type PaginatedData = {
-  [timestamp: number]: TokenTransaction[]
+  [FIRST_PAGE_CURSOR]: TokenTransaction[]
+  [endCursor: string]: TokenTransaction[]
```

Comment:
@kathaypacific We need to know for which page specifically the data has arrived. This is important for when we do polling specifically as we only need to update data for that very page and not trigger re-processing for _all_  the transactions from all the other pages. Most polling updates will return the same data over and over so re-processing all transactions on every update feels unnecessary. If everything would be pushed into a single array - there would be a need to re-filter the whole transactions array and re-merge it with the new data on every poll cycle. But if we distinguish the first page from others - we can only update the first page (we can even just replace it).

Merging the data from the page with confirmed stand by transaction happens within `updatePaginatedData` useEffect which is only triggered whenever:
- there's new data from pagination
- there's new data from confirmed stand by transactions

These are the only sources of truth that are necessary to show confirmed transactions so I think that it is completely fine to only rely on `paginatedData` to trigger recomputation of `confirmedTransactions` when the data changes.

On the other note, there's a possibility that once the page receives a new transaction - Zerion will re-compute its cursors as [per their docs](https://developers.zerion.io/reference/listwallettransactions) IDs they use in cursor are ambiguous: _"NOTE: Consider all IDs as abstract strings, without making any assumptions about their format or relying on such assumptions. There is a non-zero probability that IDs may change in the future, and this should not result in any breaking changes."_ 
This means that all the existing cursors can become irrelevant after the changes to the first page. So if we allow to update any page other than the first - there's no way for us to keep the existing cursors, for which we've already fetched the data.

The last point might be a bit more complex so please let me know if we should discuss this in a separate conversation where I can dig into more details on this.

Reply:
> These are the only sources of truth that are necessary to show confirmed transactions so I think that it is completely fine to only rely on paginatedData to trigger recomputation of confirmedTransactions when the data changes.
unfortunately i don't think this is currently the case - i don't think that `paginatedData` as a dependency array variable is triggering the effect _only_ when it changes. if i put a log statement inside the `useMemo` of `confirmedTransactions`, i can see that it's being executed on every poll interval even though i have no new transactions. so currently, i think the net effect is that we are already re-merging every poll cycle result with all known transactions. depending on the size of the transactions, this may not be a big deal though

Reply:
fwiw i'm not strongly opposed to this object type, but us to be able to reason about keeping the object keys (FIRST_PAGE, cursors) when we don't seem to be using them :)

Reply:
after walking through, my understanding is that this object type facilitates being able to differentiate between the first page results vs everything else, so that we can minimise the amount of work needed to merge standby and confirmed transactions `mergeStandByTransactionsInRange`, ty for explaining @sviderock !